app/code/Magento/ConfigurableProductGraphQl/Model/ConfigurableProductTypeResolver.php

```diff
@@ -19,7 +20,7 @@ class ConfigurableProductTypeResolver implements TypeResolverInterface
      */
     public function resolveType(array $data) : string
     {
-        if (isset($data['type_id']) && $data['type_id'] == 'configurable') {
+        if (isset($data['type_id']) && $data['type_id'] == Type::TYPE_CODE) {
             return 'ConfigurableProduct';
```

Comment:
Can you also please move this text-return into constant?

Reply:
Changes are done. Please check